### PR TITLE
readCourseData.edit 

### PR DESCRIPTION
**Pull request by ak840206**

### CODE REVIEW

File: edit readCourse.data [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import os,inspect
+from pymongo import MongoClient
+import pandas as pd
+import logging
+
+class Course:
+	
+    def __init__(self, *args, **kwargs):
+    	for attr in args:
+		setattr(slef,attr,[])
```
Comment (kis87988) [inline edit readCourse.data]: indentation?

File: edit readCourse.data [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  	kwargs = {'UID': '', 'crn': '', 'course_num':'', 'section_num':'', 'campus':'',\
```
Comment (kis87988) [inline edit readCourse.data]: I don't think this is right...
You pass `kwargs` into class. you should name is `some_variable_name=kwarg`. not assign empty value for each key.

File: edit readCourse.data [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+               	  'num_credit':'', 'course_title':'', 'days':'', 'startTime':u.split('-')[1], 'endTime':u.split('-')[2],\
```
Comment (kis87988) [inline edit readCourse.data]: avoid magic number `[1]`,`[2]`

**Comment by kis87988** [comment]
Please rename the title to relate title name. 
A title should not be a question.

File: edit readCourse.data [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		self.department = []
+		self.relatedCourse = []
+
+class Department:
+	def __init__(self, deptName):
+		self.deptName = deptName
+		self.courses = []
+
+class Lab:
+	def __init__(self, UID, days, startTime, endTime, \
```
Comment (kis87988) [inline edit readCourse.data]: kwargs?

File: edit readCourse.data [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+		self.UID = UID
+		self.days = days
+		self.startTime = startTime
+		self.endTime = endTime
+		self.startDate = startDate
+		self.endDate = endDate
+		self.instructor = instructor
+		self.location = location
+		
+class Logger:
```
Comment (kis87988) [inline edit readCourse.data]: does that necessary to have a logger class create by ourself?
Ref: https://realpython.com/python-logging/

File: edit readCourse.data [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+		def error(self,message):
+		self.logger.error(message)
+		def cri(self,message):
+		self.logger.critical(message)
+		
+		
+    
+def main():
+
+	logobj = Logger(__name__)
```
Comment (kis87988) [inline edit readCourse.data]: `log` or `logger` is better name. 
We don't have to create a `Logger` class
Try to use `logging.getLogger` function.

File: edit readCourse.data [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+	
+	###Don't forget to put all the .json file in the current dir
+	current_path = inspect.getfile(inspect.currentframe())
+	current_path = current_path.replace('\\', '/')
+	dir_name = os.path.dirname(current_path)
+	file_abs_path = os.path.abspath(dir_name) + "\\"
+  	file_name = input("Please enter YYYY_QQQQ_School (Example 2018_Winter_De_Anza):\n")
```
Comment (kis87988) [inline edit readCourse.data]: um... to be discuss, 
I think we don't have to ask user everytime

File: edit readCourse.data [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    
+def main():
+
+	logobj = Logger(__name__)
+	
+	###Don't forget to put all the .json file in the current dir
+	current_path = inspect.getfile(inspect.currentframe())
+	current_path = current_path.replace('\\', '/')
+	dir_name = os.path.dirname(current_path)
+	file_abs_path = os.path.abspath(dir_name) + "\\"
```
Comment (kis87988) [inline edit readCourse.data]: I don't think you can conjunction path by string `+` method
`os.path.join` maybe?

File: edit readCourse.data [hunk 9]
```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+#-*-coding:utf-8-*-
+
```
Comment (kis87988) [inline edit readCourse.data]: Can we put some comment/doc string on top of the source file?

File: edit readCourse.data [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+			temp_course = Course(*c, **v)
+			course_list.append(temp_course)
+			course_ineach_dept = "{}{}{}".format(temp_dept.deptName, temp_course.course_num, temp_course.course_title) 
+			if course_ineach_dept not in temp_dept.courses:
+				temp_dept.courses.append(course_ineach_dept)
+			total_course += 1
+		logger.info('department', department, "has", dep_course_num, "sections.")
+		dept_list.append(temp_dept)
+
+	print('total course:', total_course)
```
Comment (kis87988) [inline edit readCourse.data]: logger?

File: edit readCourse.data [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+	print('loaded course:', len(course_list))
```
Comment (kis87988) [inline edit readCourse.data]: logger?

File: edit readCourse.data [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+	file_abs_path += file_name
+	file_abs_path += "_courseData.json"
+	if file_abs_path:
+		with open(file_abs_path, 'r') as f:
+			course_raw_data = json.load(f)
+
+	course_list = []
+	dept_list = []
+	total_course = 0
+  	file_name = file_name.replace('_',' ')
```
Comment (kis87988) [inline edit readCourse.data]: why?

File: edit readCourse.data [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+  
+	for department in course_raw_data[file_name]["CourseData"]:
+		dep_course_num = 0
+		temp_dept = Department(department)
+		for c,v in course_raw_data[file_name]["CourseData"][department]:
```
Comment (kis87988) [inline edit readCourse.data]: what is `c` and `v`?

File: edit readCourse.data [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+			dep_course_num += 1
```
Comment (kis87988) [inline edit readCourse.data]: `dept_course_num`?
We may use full name for the variable naming.

File: edit readCourse.data [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+		with open(file_abs_path, 'r') as f:
+			course_raw_data = json.load(f)
+
+	course_list = []
+	dept_list = []
+	total_course = 0
+  	file_name = file_name.replace('_',' ')
+  
+	for department in course_raw_data[file_name]["CourseData"]:
+		dep_course_num = 0
```
Comment (kis87988) [inline edit readCourse.data]: `department_course_number` should be better than this

File: edit readCourse.data [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+		temp_dept = Department(department)
+		for c,v in course_raw_data[file_name]["CourseData"][department]:
+			dep_course_num += 1
+			temp_course = Course(*c, **v)
+			course_list.append(temp_course)
+			course_ineach_dept = "{}{}{}".format(temp_dept.deptName, temp_course.course_num, temp_course.course_title)
```
Comment (kis87988) [inline edit readCourse.data]: full name for variable name

File: edit readCourse.data [hunk 17]
```diff
@@ -0,0 +1,154 @@
...
+			if course_ineach_dept not in temp_dept.courses:
+				temp_dept.courses.append(course_ineach_dept)
+			total_course += 1
+		logger.info('department', department, "has", dep_course_num, "sections.")
+		dept_list.append(temp_dept)
+
+	print('total course:', total_course)
+	print('loaded course:', len(course_list))
+
+	username = input("Enter your username:")
```
Comment (kis87988) [inline edit readCourse.data]: for database? 
this should be in the `.env` file for system environment variables

File: edit readCourse.data [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+	password = input("Password:")
```
Comment (kis87988) [inline edit readCourse.data]: for database? 
this should be in the `.env` file for system environment variables

File: edit readCourse.data [hunk 19]
```diff
@@ -0,0 +1,154 @@
...
+	client = MongoClient("mongodb+srv://" + username +":" + password + "@fhdatimedb-jjsjm.mongodb.net/test?retryWrites=true&w=majority")
```
Comment (kis87988) [inline edit readCourse.data]: for database? 
this should be in the `.env` file for system environment variables

File: edit readCourse.data [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+	db = client.get_database('Instructor')
+	tc = db.new_test_courses
+	td = db.new_test_depts
+	for course in course_list:
+		temp_course = {
```
Comment (kis87988) [inline edit readCourse.data]: can we try to just put whole course in the `tc.insert_one(course)`?

File: edit readCourse.data [hunk 21]
```diff
@@ -0,0 +1,154 @@
...
+		dept_list.append(temp_dept)
+
+	print('total course:', total_course)
+	print('loaded course:', len(course_list))
+
+	username = input("Enter your username:")
+	password = input("Password:")
+	client = MongoClient("mongodb+srv://" + username +":" + password + "@fhdatimedb-jjsjm.mongodb.net/test?retryWrites=true&w=majority")
+	db = client.get_database('Instructor')
+	tc = db.new_test_courses
```
Comment (kis87988) [inline edit readCourse.data]: what is `tc`?

File: edit readCourse.data [hunk 22]
```diff
@@ -0,0 +1,154 @@
...
+	td = db.new_test_depts
```
Comment (kis87988) [inline edit readCourse.data]: what is `td`

File: edit readCourse.data [hunk 23]
```diff
@@ -0,0 +1,154 @@
...
+			'attribute' : course.attribute,
+		}
+		tc.insert_one(temp_course)
+
+	for dept in dept_list:
+		temp_dept = {
+			'department_name' : dept.deptName,
+			'course_list' : dept.courses
+		}
+		td.insert_one(temp_dept)
```
Comment (kis87988) [inline edit readCourse.data]: same with above

**Review by kis87988 — CHANGES_REQUESTED** [review]

File: edit readCourse.data (hunk 2 above)
Comment (ak840206) [reply edit readCourse.data]: Do you mean this?

class Course:
    def __init__(self, *args, **kwargs):
    	for attr in args:
	     setattr(slef,attr,[])
  	some_variable_name=kwarg
  	for key, value in kwargs:
        	setattr(slef, key, value)

I'm not sure its function and is it right to put here?

**Pull request closed without merging**